test/main-tests/test.js

```diff
@@ -242,4 +242,22 @@ describe('EslintValidationFilter', function describeEslintValidationFilter() {
       }
     });
   });
+
+  it('throws when `throwOnError` is set and result severity', function(done) {
+    this.timeout(0);
+
+    const promise = shouldReportErrors(FIXTURES, {
+      options: {
+        ignore: false,
+      },
+      throwOnError: true
+    })();
+
+    return promise.catch((err) => {
+      expect(err).to.be.an('error');
+      expect(err.message).to.equal('rules violation with `error` severity level');
+    });
```

Comment:
@Turbo87 I really like the idea of using `chai-as-promised`, but might it be better to explore that in a separate story around refactoring the larger test suite (which I definitely think we could stand to do anyway)?

I could refactor this block in the meantime like so:
```
  it('throws when `throwOnError` is set and result severity', function(done) {
    const promise = shouldReportErrors(FIXTURES, {
      options: {
        ignore: false,
      },
      throwOnError: true
    })();

    promise.then(
      null,
      (err) => {
        expect(err).to.be.an('error');
        expect(err.message).to.equal('rules violation with `error` severity level');
        done();
      }
    )
    .catch(err => done(err));
  });
```